docs/core/diagnostics/metrics-collection.md

```diff
@@ -123,255 +109,145 @@ Press p to pause, r to resume, q to quit.
+The following output shows the output of the app with custom callback on each measurement:
 
 ```dotnetcli
 > dotnet run
 Press any key to exit
-hats-sold recorded measurement 4
-hats-sold recorded measurement 4
-hats-sold recorded measurement 4
-hats-sold recorded measurement 4
+hats-sold recorded measurement 978
+hats-sold recorded measurement 775
+hats-sold recorded measurement 666
+hats-sold recorded measurement 66
+hats-sold recorded measurement 914
+hats-sold recorded measurement 912
 ...
 ```
 
-Let's break down what happens in the example above.
+### Explaining the sample code
 
-```csharp
-using MeterListener meterListener = new MeterListener();
-```
+The code snippets in this section come from the preceding sample.
 
-First we created an instance of the <xref:System.Diagnostics.Metrics.MeterListener>, which we will use to receive measurements.
+In the following highlighted code, an instance of the <xref:System.Diagnostics.Metrics.MeterListener> is created to receive measurements. The `using` keyword causes `Dispose` to be called when the `meterListener` goes out of scope.
 
-```csharp
-meterListener.InstrumentPublished = (instrument, listener) =>
-{
-    if(instrument.Meter.Name == "HatCo.HatStore")
-    {
-        listener.EnableMeasurementEvents(instrument);
-    }
-};
-```
+:::code language="csharp" source="snippets/Metrics/Program.cs" id="snippet_uml" highlight="1":::
+
+The following highlighted code configures which instruments the listener receives measurements from. <xref:System.Diagnostics.Metrics.MeterListener.InstrumentPublished> is a delegate that is invoked when a new instrument is created within the app.
 
-Here we configured which instruments the listener will receive measurements from.
-<xref:System.Diagnostics.Metrics.MeterListener.InstrumentPublished> is a delegate that will be invoked anytime a new
-instrument is created within the app. Our delegate can examine the instrument, such as checking the name, the Meter, or any other
-public property to decide whether to subscribe. If we do want to receive measurements from this instrument, then we invoke
-<xref:System.Diagnostics.Metrics.MeterListener.EnableMeasurementEvents%2A> to indicate that. If your code has another way
-to obtain a reference to an instrument, it's legal to invoke `EnableMeasurementEvents()` at any time with that reference, but this is
-probably uncommon.
+:::code language="csharp" source="snippets/Metrics/Program.cs" id="snippet_uml" highlight="2-99":::
 
+The delegate can examine the instrument to decide whether to subscribe. For example, the delegate can check the name, the Meter, or any other public property. <!-- The previous version was way too long to MT. Is this better? --> <xref:System.Diagnostics.Metrics.MeterListener.EnableMeasurementEvents%2A> enables receiving measurements from the specified instrument.<!-- recommend deleting the following--> Code that obtains a reference to an instrument by another approach:
+
```

Comment:
Deleting the text about "Code that obtains a reference to an instrument by another approach..." is fine.